Documentation/ClangdConfiguration.md

```diff
@@ -0,0 +1,50 @@
+# Configuring the clangd Language Server
+
+clangd is recommended as a C++ language server in general, as it is most likely to work on all platforms.
+
+CLion ships with a special build of clangd that usually works fine. Most other editors can be configured to use any build of clangd via the LSP or dedicated extensions/plugins. Refer to the editor-specific documentation on how to install and enable such extensions.
+
+## `.clangd`
+
+clangd uses `compile_commands.json` files to understand the project. CMake will generate these in directories such as Build/x86_64, Build/x86_64clang, and Build/lagom.
+Depending on which configuration (architecture and toolchain) you use most, set the CompilationDatabase configuration item in the below `.clangd` file accordingly.
+
+The recommended `.clangd` file (at the root of your Serenity checkout) is as follows; it should work out of the box and can be adjusted as needed:
+
+```yaml
+CompileFlags:
+    # Treat code as if we’re compiling for Serenity (and not Lagom)
+    Add: [-D__serenity__]
+    # You can remove unwanted flags such as those that aren't supported by the current version of clang.
+    Remove: []
+    CompilationDatabase: Build/x86_64
+
+Style:
+    # clangd 20+: Use correct include style.
+    AngledHeaders: ["AK/.*", "Userland/.*", "Kernel/.*", "Applications/.*", "Lib.*/.*"]
+
+Diagnostics:
+    UnusedIncludes: None
+    MissingIncludes: None
```

Comment:
We might be able to turn these up to `Strict` and `Loose` respectively

Reply:
Seems to work fine, but not sure if I even have files where this applies.

Reply:
Oh just looked at the docs again, seems there is only `None` or `Strict`
and without `AnalyzeAngledIncludes` they won't actually do much
The problem with `MissingIncludes: Strict` though is that it highlights all tokens which aren't directly included, not sure if we can disable